[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/repos-especializacion-UdeA/estadistica/blob/main/trabajo1.ipynb)

In [2]:
import scipy.stats as stats
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Trabajo 1

Para solucionar los siguentes enunciados es necesario platear, de ser posible, la(s) variable(s) a utilizar, su distribución y los parámetros con sus respectivos valores. Además su solución debe incluir el código en Python utilzado para su solución.

## Integrantes

* Jairo Agudelo.
* Henry Arcila.

## Problemas

1. Un fabricante asegura que solo el 5% de su producción total se encuentra defectuosa. Supóngase que se ordenan 1000 artículos y se seleccionan 25 al azar para inspeccionarlos. Si el fabricante se encuentra en lo correcto, la probabilidad de observar dos o más artículos defectuosos en la muestra es?

   **Solucion**:

   Sea $X$ el numero de articulos defectuosos tomados de la selección de 25 elementos. Tenemos que $X \sim Binom(n,p)$ donde:
   * $n = 25$
   * $p = 0.05$

   Y lo que se pide es: $P(X \geq 2) = ?$

In [11]:
"""
Definición de la VA binomial: X ~ Bin(n,p)

X: el numero de articulos defectuosos tomados de la selección de 25 elementos
- n = 25
- p: 0.05
- q: 1 - p = 0.95
"""
X = stats.binom(n = 25, p = 0.05) # X ~ Binom(n = 25, p = 0.05)
P_lt_1 = X.cdf(1)
P_geq_2 = 1 - P_lt_1
print(f"P(X <= 2) = {P_geq_2:.5}")


P(X <= 2) = 0.35762


**Rta**: La probabilidad de observar dos o mas articulos defectuosos en la muestra es de un 35.76%

2. Para los estudiantes es de vital importancia tener un almuerzo tranquilo, pero fundamentalmente que sea servido con rapidez. En una universidad hay dos restaurantes grandes que sirven ricos platos, el primer restaurante se demora en promedio en atender 20 minutos con una desviación de 3 minutos, el segundo tarda en promedio 17 minutos en atender con una desviación de 8 minutos. Suponga que los tiempos en los que se atiende en ambos restaurantes se distribuyen de manera normal. Un estudiante de particular cuenta con un máximo de 21 min para ser atendido. ¿El restaurante a elegir es?
   
   **Solucion**:

   Sea $X_i$ el tiempo de atencion (en min) del restaurante $i$. Con $i = {1,2}$. Ademas sea $X_i \sim N(\mu_i, \sigma_i^2)$; para cada restaurante:
   * **Restaurante 1**:
     * $\mu_1 = 20$
     * $\sigma_1 = 3$
   * **Restaurante 2**:
     * $\mu_2 = 17$
     * $\sigma_2 = 8$

   Para responder la pregunta tenemos que calcular $P(X_1 <= 21)$ y $P(X_2 <= 21)$ y el valor que de mayor de las probabilidades anteriores será el que determine el restaurante que se elegirá:

In [12]:
"""
Restaurante 1: X1 ~ N(mu1,sigma1)

- mu1 = 20
- sigma1 = 3
"""

mu1 = 20
sigma1 = 3

X1 = stats.norm(loc = mu1, scale = sigma1)

P1_leq_21 = X1.cdf(21)
print(f"P(X1 <= 12) = {P1_leq_21:.5}")

"""
Restaurante 2: X2 ~ N(mu2,sigma2)

- mu2 = 17
- sigma2 = 8
"""

mu2 = 17
sigma2 = 8

X2 = stats.norm(loc = mu2, scale = sigma2)
P2_leq_21 = X2.cdf(21)
print(f"P(X2 <= 12) = {P2_leq_21:.5}")

if P2_leq_21 == P1_leq_21:
    print("Da igual")
elif P1_leq_21 > P2_leq_21:
    print("La mejor opcion es el restaurante 1")
else:
    print("La mejor opcion es el restaurante 2")

P(X1 <= 12) = 0.63056
P(X2 <= 12) = 0.69146
La mejor opcion es el restaurante 2


**Rta**: La mejor opción para comer el almuerzo es el restaurante 2

3. El tiempo entre llegadas de avionetas a un aeropuerto tiene una distribución exponencial con un tiempo medio de 12 horas. Si se escogen los siguientes 25 intervalos de una hora, la probabilidad de que exactamente en uno de ellos, hayan aterrizado más de tres avionetas, ¿es?
   
   * **Solución**:
     
     Este problema se resuelve en dos partes. 
     1. Uso de una distribución de Poisson para determinar la probabilidad de que en una hora hayan aterrizado mas de tres avionetas. Esto es:
        
        Sea $X$ el número de avionetas que aterrizan en un intervalo de una hora. Con $X \sim Poisson(\lambda = 1/12)$. Tenemos:
        * $\lambda = 1/12$
        * $P(X > 3) = ?$

     2. Uso de una distribución Binomial para saber el numero de intervalos en los que se cumple que aterrizan mas de tres avionetas.
        
        Sea $Y$ el numero de intervalos (de un total de 25) en los que aterrizan mas de 3 avionetas. Con $Y \sim Binom(n = 15, p = P(X \gt 3)). Esto es:
        * $n = 25$
        * $p = P(X \gt 3)$, donde el valor de $P(X > 3)$ se obtiene de la primera parte.
        * $P(Y = 1) = ?$

     Al final del problema, lo que se pide es $P(Y = 1) = ?$

In [24]:
# ---------------------------- Parte 1 ----------------------------
l = 1/12 # lambda

X = stats.poisson(mu = l) # Poison: X ~ Poisson(lambda = 1/12)
P_gt_3 = 1 - X.cdf(3)
print(f"P(X > 3) = {P_gt_3:.8f}")

# ---------------------------- Parte 1 ----------------------------
n = 25
p = P_gt_3
Y = stats.binom(n = 25, p = P_gt_3) # X ~ Binom(n = 25, p = 0.05)
P_eq_1 = Y.pmf(1)
print(f"P(Y = 1) = {P_eq_1:.6f}")

P(X > 3) = 0.00000188
P(Y = 1) = 0.000047


**Rta**: La probabilidad de que haya más de tres llegadas en un exactamente uno de los 25 intervalos es de 0.0047% lo cual hace que este evento sea muy poco probable.

4. Por experiencia se sabe que el 25% de los estudiantes de un curso, el cual se califica de 0 a 5, lo reprueban. Se sabe que las notas obtenidas por los estudiantes en dicho curso tienen una distribución normal con una nota media de 3.6. ¿Si un estudiante termina el curso, la probabilidad de que su nota sea superior a 4 es?
   
   **Solución**:

   Sea $X$ la calificación en el rango [0,5] obtenida por el estudiante. Según el enunciado:
   * $X \sim N(\mu = 3.6, \sigma^2 = ?)$ 
   * $P(X < 3) = 0.25$
   
   Se pide: $P(X>4) = ?$

In [17]:
mu = 3.6
P_lt_3 = 0.25

# Obtencion del percentil valor Z asociado al 25%
Z = stats.norm.ppf(P_lt_3)

# Obteniendo el valor de la desviacion estandar
sigma = -0.6/Z
print(f"sigma = {sigma:.4}")

"""
Datos asociados a la VA normal

X ~ N(mu = 3.6, sigma = 0.8866)
"""
X = stats.norm(loc = mu, scale = sigma)

# Probabilidad pedida
P_geq_4 = 1 -  X.cdf(4)
print(f"P(X >= 4) = {P_geq_4:.5}")


sigma = 0.8896
P(X >= 4) = 0.32648


**Rta**: La probabilidad de que un estudiante saque mas de 4 al terminar el curso es de un 32.648 %

5. El operador de una estación de bombeo observa que el tiempo de consumo del agua durante las primeras horas de la tarde tiene aproximadamente una distribución exponencial con una media de 120 seg. La probabilidad de que el tiempo de consumo sea superior a los 180 seg. durante las primeras horas de la tarde en un día seleccionado al azar es?
   
   **Solución**:

   Sea $X$ el tiempo de consumo en segundos. Ademas tenemos que:
   * $E(X) = 120$
   * $\lambda = 1/E(X) = 1/120$ 
   
   Luego $X \sim Exp(\lambda = 1/20)$ y se pide: $P(X > 180) = ?$ 


In [19]:
"""
Definición de la VA exponencial: X ~ Exp(lambda = 1/20)
"""
E = 120
l = 1/120

X = stats.expon(scale = 1/l)

P_gt_180 = 1 - X.cdf(180)
print(f"P(X > 180) = {P_gt_180:.5}")

P(X > 180) = 0.22313


**Rta**: La probabilidad de que el tiempo de consumo sea superior a los 180 segundos es de 22.313 %